src/resilience/context/Context.cpp

```diff
@@ -70,7 +74,48 @@ namespace KokkosResilience
   std::unique_ptr< ContextBase >
   make_context( const std::string &config )
   {
-    auto cfg = Config{ config };
-    return std::unique_ptr< ContextBase >{};
+    return make_context( Config{ config } );
   }
+
+  std::unique_ptr< ContextBase >
+  make_context( const Config &cfg )
+  {
+    using fun_type = std::function<std::unique_ptr<ContextBase>()>;
+    static std::unordered_map<std::string, fun_type> backends = {
+        {"stdfile", [&]() -> std::unique_ptr<ContextBase> {
+           return std::make_unique<StdFileContext<StdFileBackend> >(cfg);
+         }}};
+  
+    auto pos = backends.find(cfg["backend"].as<std::string>());
+    if (pos == backends.end()) return {};
+  
+    return pos->second();
+  }
+
+#ifdef KR_ENABLE_MPI_CONTEXT
```

Comment:
This will be simpler once the backends are virtual, but for now:
```suggestion
#if defined(KR_ENABLE_MPI_CONTEXT) && defined(KR_ENABLE_VELOC_BACKEND)
```

Reply:
I decided to move thse down into the function body -- basically it's still valid to call make_context but it will return a nullptr as there would not be any supported MPI backends if VeloC is not enabled

Reply:
e.g.

```cpp
#ifdef KR_ENABLE_MPI_CONTEXT
        {"veloc", [&]() -> std::unique_ptr<ContextBase> {
          return std::make_unique<MPIContext<VeloCMemoryBackend> >(comm, cfg);
        }},
        {"veloc-noop", [&]() -> std::unique_ptr<ContextBase> {
          return std::make_unique<MPIContext<VeloCRegisterOnlyBackend> >(comm, cfg);
        }}
#endif
```